.github/workflows/reusable_data_model_gen.yml

```diff
@@ -0,0 +1,67 @@
+name: Reusable Data Model Generator
+
+on:
+  workflow_dispatch:
+
+jobs:
+  vars:
+    name: Set Variables
+    outputs:
+      user: ${{ steps.data.outputs.user }}
+      pass: ${{ steps.data.outputs.pass }}
+    runs-on: ubuntu-22.04
+    timeout-minutes: 1
+    steps:
+      - name: Generate random username and password
+        id: data
+        run: |
+          USERNAME="user$(shuf -i 1000-9999 -n 1)"
+          PASSWORD=$(openssl rand -base64 12)
+
+          echo "user=$USERNAME" >> $GITHUB_OUTPUT
+          echo "pass=$PASSWORD" >> $GITHUB_OUTPUT
+
+  schemaspy:
+    name: Generate Documentation
+    runs-on: ubuntu-22.04
+    needs: [vars]
+    services:
+      postgres:
+        image: postgis/postgis:16-3.4
+        env:
+          POSTGRES_DB: fam
+          POSTGRES_USER: ${{ needs.vars.outputs.user }}
+          POSTGRES_PASSWORD: ${{ needs.vars.outputs.pass }}
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+        ports:
+          - 5432:5432
+    timeout-minutes: 10
+    steps:
+      - uses: actions/checkout@v4
+      - name: Flyway
+        uses: docker://flyway/flyway:10
+        env:
+          FLYWAY_URL: jdbc:postgresql://postgres:5432/fam
+          FLYWAY_USER: ${{ needs.vars.outputs.user }}
+          FLYWAY_PASSWORD: ${{ needs.vars.outputs.pass }}
+          FLYWAY_LOCATIONS: filesystem:./server/flyway/sql
+          FLYWAY_DEFAULT_SCHEMA: "app_fam"
```

Comment:
Database name is "fam" to match our docker (shouldn't matter though, as long as we used the same name below when run the flyway. In terraform we deployed our db named "famdb"). 

Database schema is "app_fam"

username and password as long as match the ones we used when create the database will be good